File: android/app/src/main/java/com/example/idea_book/domain/model/IdeaModel.kt
package com.example.idea_book.domain.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

data class IdeaModel(
    val id: Int,
    val title: String,
    val description: String,
    val username: String,
    val user_id: String,
    val likes_total: Int,
    val comments_total: Int,
    val tags: List<TagModel>,
    val likes: List<String>,
    val created_at: String,
)

data class TagModel(
    val id: Int,
    val name: String,
    val icon: String
)

File: server/app/models/IdeaModel.go
package models

import "gorm.io/gorm"

type Idea struct {
	gorm.Model
	Title         string    `json:"title"`
	Description   string    `json:"description"`
	UserUid       string    `json:"user_id"`
	Username      string    `json:"username"`
	Likes         []Like    `json:"likes"`
	Comments      []Comment `json:"comments"`
	CommentsTotal int       `json:"comments_total"`
	LikesTotal    int       `json:"likes_total"`
	Tags          []Tag     `json:"tags" gorm:"many2many:idea_tags;"`
}
